fe/fe-core/src/test/java/org/apache/doris/nereids/memo/MemoExploreTest.java

```diff
@@ -0,0 +1,148 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.memo;
+
+import org.apache.doris.common.jmockit.Deencapsulation;
+import org.apache.doris.nereids.properties.LogicalProperties;
+import org.apache.doris.nereids.properties.PhysicalProperties;
+import org.apache.doris.nereids.trees.plans.FakePlan;
+import org.apache.doris.nereids.trees.plans.JoinType;
+import org.apache.doris.nereids.trees.plans.logical.LogicalJoin;
+import org.apache.doris.nereids.trees.plans.logical.LogicalOlapScan;
+import org.apache.doris.nereids.trees.plans.logical.LogicalProject;
+import org.apache.doris.nereids.util.MemoTestUtils;
+import org.apache.doris.nereids.util.PatternMatchSupported;
+import org.apache.doris.nereids.util.PlanChecker;
+import org.apache.doris.nereids.util.PlanConstructor;
+import org.apache.doris.qe.ConnectContext;
+
+import com.google.common.collect.Lists;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+class MemoExploreTest implements PatternMatchSupported {
+
+    private final ConnectContext connectContext = MemoTestUtils.createConnectContext();
+
+    private final LogicalJoin<LogicalOlapScan, LogicalOlapScan> logicalJoinAB = new LogicalJoin<>(JoinType.INNER_JOIN,
+            PlanConstructor.newLogicalOlapScan(0, "A", 0),
+            PlanConstructor.newLogicalOlapScan(1, "B", 0));
+
+    /*
+     * ┌─────────────────────────┐     ┌───────────┐
+     * │  ┌─────┐       ┌─────┐  │     │  ┌─────┐  │
+     * │  │0┌─┐ │       │1┌─┐ │  │     │  │1┌─┐ │  │
+     * │  │ └┼┘ │       │ └┼┘ │  │     │  │ └┼┘ │  │
+     * │  └──┼──┘       └──┼──┘  │     │  └──┼──┘  │
+     * │Memo │             │     ├────►│Memo │     │
+     * │  ┌──▼──┐       ┌──▼──┐  │     │  ┌──▼──┐  │
+     * │  │ src │       │ dst │  │     │  │ dst │  │
+     * │  │2    │       │3    │  │     │  │3    │  │
+     * │  └─────┘       └─────┘  │     │  └─────┘  │
+     * └─────────────────────────┘     └───────────┘
+     */
+    @Test
+    void test() {
```

Comment:
rename test into testMergeGroup

Reply:
There will more test case in the future, we can rename in add new case.